### ISSUE TITLE
Remove DurableTask logic from SyncTriggers

### ISSUE THREAD

**Issue by mathewc** [Azure/azure-functions-host#5390]
We need to find a way to remove the DurableTask specific logic from the SyncTriggers path. Some tactical work was done in https://github.com/Azure/azure-functions-host/pull/5282 as a stopgap, but once we make the required ScaleController changes to enable it to determine itself the version of Durable being used and the right TaskHubName to use, we can remove all this from the runtime.

**Comment by cgillum** [comment]
@mathewc for my own curiosity, what's the motivation for removing this logic? Is it just to make the code cleaner or is there a more pressing reason? I ask because the Scale Controller definitely needs this information (and probably more than this).

I'm actually discussing some scenarios with Paul where we might want host.json config information (e.g. max concurrency config) for other triggers besides Durable Functions. If we could come up with a way to share host.json with the Scale Controller in a general way (not specific to Durable) then that might be a great alternative to what we're doing today.

**Comment by mathewc** [comment]
The functions host shouldn't have any extension specific logic, references, etc. We strive to keep the design clean so extensions are pluggable without any coupling. This ensures that extension capabilities are available to all extensions w/o requiring any host updates.

**Comment by ConnorMcMahon** [comment]
Is there a reason the SyncTriggers api response couldn't just include the host.json (or at least the extensions + concurrency knobs) so that the ScaleController has any information it could need?

In general, more and more trigger types are relying on common configuration that lives in the host.json, and adding specific logic for each of them to the SyncTriggers API is not super maintainable.

**Comment by mathewc** [comment]
Yes, we can extend the sync payload to include additional info like that. When I extended the payload format a while ago to the current ```{ triggers: [], functions: [], secrets: {} }``` format (for the ARM cache work), I set it up for future extensions. So yes something like that would be preferable to extensions duplicating host level info in each function object in the payload.

**Comment by davidmrdavid** [comment]
This seems worth revisiting given the bug we found here: https://github.com/Azure/azure-functions-host/pull/9331
FYI @bachuv.

I tend to agree that it would be best for the Host not to maintain Durable Functions specific logic, like it does today for the SyncTriggers operation. I see two paths moving forward:
(1) To simply provide the host.json config as part of SyncTriggers. I think this would be the easiest approach
(2) To allow the extension to intercept the SyncTriggers payload and provide it with extra metadata as needed. This might be the more general solution. I would prefer this as it gives extensions more control.

@mathewc: Do we have the means to implement something like option (2) today?

**Comment by mathewc** [comment]
Update on this: We did work in [this PR](https://github.com/Azure/azure-functions-host/pull/9898) to always include the `hostConfig` section in sync triggers payloads. That information includes all the info needed by scale controller for scaling. The initial PR [here](https://github.com/Azure/azure-functions-host/pull/8148) describes the format.

So to @davidmrdavid's comment above, I think we might have what we need now to address this. Thoughts @AnatoliB ?

**Comment by AnatoliB** [comment]
@cgillum ?

**Comment by cgillum** [comment]
As long as the scale controller continues to work, I'm good with this change, and happy to remove trigger-specific logic.

**Comment by mathewc** [comment]
Some more notes on what I think needs to be done here:

1. The [DurableTaskOptions](https://github.com/Azure/azure-functions-durable-extension/blob/f0818f8fe128b1084ce6e56752c8d0cb399e1952/src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs#L19) type needs to be updated to implement IOptionsFormatter. See any of the other extension options types, e.g. [EventHubOptions](https://github.com/Azure/azure-sdk-for-net/blob/9161f2fe42b210052774a80a130f59b8619ecba0/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs#L17). This ensures that the options values you need are included by the code [here](https://github.com/Azure/azure-functions-host/blob/f44cd1f192d89b42ed6e8f01658b9bc517ee0db6/src/WebJobs.Script/HostOptionsProvider.cs#L62) when sync triggers is performed. Ultimately, all the options gathered in this way from the various extensions are included into the sync payload as a "hostConfig" section, by code [here](https://github.com/Azure/azure-functions-host/blob/f44cd1f192d89b42ed6e8f01658b9bc517ee0db6/src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs#L323). In addition, the IOptionsFormatter changes improve servicing since such options are also logged on startup. 
2. ScaleController then needs to be updated to use this. It reads hostConfig [here](https://msazure.visualstudio.com/One/_git/AAPT-Antares-ScaleController?path=/src/ScaleController/SiteMetadata.cs&version=GBmain&_a=contents).
3. After this, the durable specific code in the host [here](https://github.com/Azure/azure-functions-host/blob/dc299b1b1459d599709e2bce9760b81fca3e4d82/src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs#L499) that is reading the Durable host.json section and copying/duplicating that onto all triggers can be removed since ScaleController will have that info itself now.

Of course in the above, things need to be staged such that we give time for the Durable extension updates to roll out completely, for all users to move onto the new version, for sync trigger operations to happen to update the cached triggers for apps so that ScaleController will indeed have the info it needs.
